### ISSUE TITLE
PC Utility test run for Lidar testing and lidar scanning 

### ISSUE THREAD

**Issue by AbidHussain70** [iliasam/OpenTOFLidar#5]
I did my first run but I am not getting the distance measurement. Any clue?
![GUI-for-github](https://user-images.githubusercontent.com/23444211/81367012-a3b7ce80-911e-11ea-8938-d0cd829d5a0e.jpg)

and I also changed encodre count to 20,
My encoder gives digital output 3v.

![GUI2-for-github](https://user-images.githubusercontent.com/23444211/81367122-ee394b00-911e-11ea-9393-dc5e084a2601.jpg)

![GUI3-for-github](https://user-images.githubusercontent.com/23444211/81367148-ff825780-911e-11ea-8546-340c394cda4d.jpg)

**Comment by iliasam** [comment]
Have you tried to follow  this instructions: https://github.com/iliasam/OpenTOFLidar/wiki/Electronics-assembling-notes ?
Firstly, you should test your device in not-scanning mode: https://github.com/iliasam/OpenTOFLidar/tree/develop/Firmware
Do not try to run scanning mode until you get good results in not-scanning mode.
  
I can see from the first image that the is no received signal or it is too low.
Also I can see that you set APD target voltage to 132V, but measured voltage is 126V.
It would be good to measure breakdown voltage of the APD.
For AD500-8 it is close to 90V, for MTAPD-07-13 it is close to 130V.
But for testing you could set less voltage - to avoid breakdown .

**Comment by AbidHussain70** [comment]
Great, it worked for testing utility app. 

![Distance-getting](https://user-images.githubusercontent.com/23444211/81464685-5f97fd00-91f6-11ea-9e1f-0c9461538f2c.png)

![Scanner](https://user-images.githubusercontent.com/23444211/81464770-13998800-91f7-11ea-9cf1-0f28d4846505.jpg)

**Comment by iliasam** [comment]
I can see from the histogram that received signal is still very small. See  this page with examples: https://github.com/iliasam/OpenTOFLidar/wiki/Resolution-of-the-LIDAR
Do you have an oscilloscope to  see received signal form?

**Comment by iliasam** [comment]
Sorry, previous link is for scanning mode.
You should see screenshots of the "Testing Utility".
You can find them here: https://github.com/iliasam/OpenTOFLidar/wiki/LidarTestingUtility
Here: https://ibb.co/vjFfwVD
And in the video: https://youtu.be/lTPH_Xa9yCk

**Comment by iliasam** [comment]
Should I close this issue?

**Comment by AbidHussain70** [comment]
Hi,
I will do the oscilloscope test and I will post the results.

**Comment by AbidHussain70** [comment]
I checked with oscilloscope, with changing Vlaser, the Vpp on connector J2 changes. The width of the pulse also increases with increase in Vlaser. 

> Vlaser 16V, output is below.

.
![scanner-16V](https://user-images.githubusercontent.com/23444211/82534882-8bad6980-9b78-11ea-9aff-f985f6b23c02.jpg)

> Vlaser 15V oscilloscope result is given below.
![scanner-15V](https://user-images.githubusercontent.com/23444211/82535099-f5c60e80-9b78-11ea-8009-c1db2c26ff47.jpg)

**Comment by AbidHussain70** [comment]
The signal strength increases on white surface. But on Random surfaces the strength remains the same. By same I mean the same as shown in the pictures I posted earlier.

**Comment by iliasam** [comment]
If you oscilloscope configuration is right, than you pictures mean that laser current is too low.
At the first picture one cell is 50mV, so signal peak amplitude is ~150-200mV. Current sensing resistor must be 0.1 ohm, so current is ~1.5-2A. This is too low.
See this picture for a reference: https://github.com/iliasam/OpenTOFLidar/blob/develop/Images/Oscilloscope/LaserCurrent.png
Could you say your oscilloscope model?
Is laser voltage really 15-16V ?

**Comment by AbidHussain70** [comment]
Sorry I forgot to mention, my oscilloscope probe is 10x. 

> ~150-200mV means ~1.5-2V. 

>

**Comment by iliasam** [comment]
Ok, but what about strong additional noise?
Could you measure TIA signal at J1 ? 
It would be good to measure signals at J1 and J2 simultaneously.

**Comment by AbidHussain70** [comment]
I order the connectors, I will test it once I received the connectors.

**Comment by AbidHussain70** [comment]
I checked but I am not getting any pulse at TIA J1.

**Comment by iliasam** [comment]
Have you tried to check if laser is really working with a phone camera?
It is strange that you can't see any pulses, while you can get some measurements (as you showed earlier).
Amplitude of the pulses should be low - see here: https://ibb.co/m9Yj5TD
It is useful to use J2 laser signal as a trigger for the oscilloscope.

**Comment by AbidHussain70** [comment]
![laser-camera-test](https://user-images.githubusercontent.com/23444211/84105966-75dee600-aa4c-11ea-9a25-fe9eb1218043.jpg)

This is laser check with camera. But I noticed it is blinking in irregular interval.

**Comment by iliasam** [comment]
> But I noticed it is blinking in irregular interval.
I think that this is OK. It could be produced by camera frame rate.

Could you confirm that you can see that "START" values are changing somehow when you are changing distance to the object (as you said earlier)?
Also, I advice you to make non-scanning tests without mirror (to avoid possible wrong reflections).

**Comment by AbidHussain70** [comment]
![OpenTOFLidar-Testing-Utility-v10](https://user-images.githubusercontent.com/23444211/84238213-43f18080-ab2d-11ea-88e7-695aa3a4195e.gif)

I am getting the distance in meters when I move an object away from the scanner. But the measurement is not stable. And also accuracy is not good.

**Comment by AbidHussain70** [comment]
I will assemble another scanner.

**Comment by iliasam** [comment]
I can see that your photodiode is really receiving light signal and TIA is working somehow.
So it very strange that you can't see TIA signal at the oscilloscope.
Tip: you must use AC coupling it the oscilloscope.
There are two possible reasons why your received signal is low:
1. Laser is not good aligned. Try to move laser holder against PCB.
2. TIA is particularly damaged. I have seen such case in my practice.

**Comment by AbidHussain70** [comment]
![WeChat Photo Editor_20200610160516](https://user-images.githubusercontent.com/23444211/84243075-44414a00-ab34-11ea-9815-7a36126c52cf.jpg)
This is what I got when both J1 and J2 is connected with oscilloscope.

**Comment by iliasam** [comment]
Is oscilloscope CH2 divider correct?
TIA have ~100mV overdrive limit, so maximum CH2 should be ~100mV.

**Comment by AbidHussain70** [comment]
![WeChat Image_20200610164054](https://user-images.githubusercontent.com/23444211/84246438-36da8e80-ab39-11ea-93c9-102e26d3b86b.jpg)

![WeChat Image_20200610164105](https://user-images.githubusercontent.com/23444211/84246466-3fcb6000-ab39-11ea-8608-eea3acde81a8.jpg)


**I am not using probe I am using BNC connectors.** 
I got the pulse. Problem was with the UFL connectors.

**Comment by iliasam** [comment]
Now it is looking good.
My tests are looking so: https://ibb.co/m9Yj5TD
You can see that at strong overdrive signal length should be > 75 ns

**Comment by AbidHussain70** [comment]
![TIA-PULSE](https://user-images.githubusercontent.com/23444211/84250059-20830180-ab3e-11ea-9fb9-be7371f0bce6.gif)

**Comment by AbidHussain70** [comment]
How should I update the calibration file? 

I think low values in bin graph is because of calibration.

**Comment by iliasam** [comment]
Calibration data is needed for "CORR. START" values.
Histogram is callculated by "START" column values.
See https://github.com/iliasam/OpenTOFLidar/wiki/LidarTestingUtility - calibration is partially described there. 
You need to record data - press START button, change received amplitude by slowly closing APD lens, press STOP.
After that you need to process data at the "Calibration" page.

**Comment by iliasam** [comment]
I added addition information to the Wiki about testing PCB without optics, it could be useful for you: https://github.com/iliasam/OpenTOFLidar/wiki/No-optics-tests

**Comment by iliasam** [comment]
Do you have any progress with this problem?

**Comment by AbidHussain70** [comment]
Hi, 
I will post an update soon.

**Comment by AbidHussain70** [comment]
I did no optics test. 
**Actual distance is 0.8 meter.**  **white wall.** 
<img width="951" alt="without-optics-distance-1 2meter" src="https://user-images.githubusercontent.com/23444211/85254389-c6b1fe00-b492-11ea-805a-42ff5ff62315.PNG">
**Laser voltage is 16V.**

<img width="957" alt="without-optics-distance-1 2meter-15v" src="https://user-images.githubusercontent.com/23444211/85254489-037df500-b493-11ea-8e41-e1f80e41b776.PNG">
**laser voltage is 15v Actual distance is 0.8 meter. white wall.**

<img width="948" alt="without-optics-distance-1 2meter-14v" src="https://user-images.githubusercontent.com/23444211/85254631-535cbc00-b493-11ea-8471-7beed2062426.PNG">

**laser voltage is 14v Actual distance is 0.8 meter. white wall**

<img width="960" alt="without-optics-distance-1 2meter-13v" src="https://user-images.githubusercontent.com/23444211/85254719-7edfa680-b493-11ea-91d8-217082f8b310.PNG">

**laser voltage is 13v Actual distance is 0.8 meter meter. white wall**

<img width="960" alt="without-optics-distance-1 2meter-12v" src="https://user-images.githubusercontent.com/23444211/85254787-a767a080-b493-11ea-841a-f867a871a2c9.PNG">

**laser voltage is 12v Actual distance is 0.8 meter. white wall**

![Uploading without-optics-distance-1.2meter-11v.PNG…]()

**laser voltage is 11v Actual distance is 0.8 meter meter. white wall**

<img width="956" alt="without-optics-distance-1 2meter-10v" src="https://user-images.githubusercontent.com/23444211/85255054-17762680-b494-11ea-9356-d8e2cb982246.PNG">

**laser voltage is 10v Actual distance is 0.8 meter. white wall**

**Comment by AbidHussain70** [comment]
For ADP test what does "TIA positive output signal amplitude" mean. I check Vpp with different ADP voltage but i did not notice any change in Vpp of the acquired pulse.

**Comment by iliasam** [comment]
It's an amplitude of TIA pulse. It's changing with no optics like this: https://ibb.co/m9Yj5TD

**Comment by AbidHussain70** [comment]
In my case just pulse width is changing not amplitude.

**Comment by AbidHussain70** [comment]
but with optics signal amplitude changes when I change the APD voltage.

**Comment by iliasam** [comment]
"In my case just pulse width is changing not amplitude."
Maybe something reflecting light from the laser?
When I say - "no optics", I mean that lens holders are installed (without lens).

**Comment by AbidHussain70** [comment]
I just remove the optics part not the holder.

**Comment by iliasam** [comment]
Maybe APD voltage is too high?

**Comment by iliasam** [comment]
How are your results?

**Comment by Anesthesia-RW** [comment]
Hi,the non-scan is working but In the scan mode, I get the following result. No point in my result and the state  mask is  0x40. If you give some suggestions I will very appreciation.
![fb](https://user-images.githubusercontent.com/48154419/90618292-59c4a580-e242-11ea-956e-9fc62c6a5683.PNG)

**Comment by Anesthesia-RW** [comment]
when i set the APD volt ,there are no change that means the "set" command don't work?
I check the src ,the state mask is 0x40 means there have no CALIBRATION . 
so in scanning model the callbration is what?

**Comment by iliasam** [comment]
See info about calibration at this page: https://github.com/iliasam/OpenTOFLidar/wiki/LidarTestingUtility
You need to record data with constant distance, but changing amplitude.

"when i set the APD volt ,there are no change that means the "set" command don't work?"
I'm sorry, I need more information about that you are doing.

**Comment by Anesthesia-RW** [comment]
Okay, I will try a few more times, and then record more details of the problem.

**Comment by Anesthesia-RW** [comment]
Hi，my test data as follow picture:
**Actual distance: 0.22m .  white wall.**
---------------------------------------------------------------------------------
Laser Volt=16V
![1](https://user-images.githubusercontent.com/48154419/90977508-952fde80-e578-11ea-9668-d451c4ca3d72.PNG)
-------------------------------------------------------------------------------
Laser Volt=15V
![2](https://user-images.githubusercontent.com/48154419/90977510-96f9a200-e578-11ea-93a5-b9a283558a2c.PNG)
------------------------------------------------------------------------------
Laser Volt=14V
![3](https://user-images.githubusercontent.com/48154419/90977512-982acf00-e578-11ea-9c61-9a51eacfecd4.PNG)
------------------------------------------------------------------------------
Laser Volt=13V
![4](https://user-images.githubusercontent.com/48154419/90977516-99f49280-e578-11ea-88b4-ad33b29463e9.PNG)
------------------------------------------------------------------------------
Laser Volt=12V
![5](https://user-images.githubusercontent.com/48154419/90977518-9b25bf80-e578-11ea-9679-fb5cbb6b2bdf.PNG)
------------------------------------------------------------------------------
Laser Volt=11V
![6](https://user-images.githubusercontent.com/48154419/90977520-9c56ec80-e578-11ea-9104-bc16c0082f09.PNG)
------------------------------------------------------------------------------
Laser Volt=10V
![7](https://user-images.githubusercontent.com/48154419/90977522-9d881980-e578-11ea-97a0-7f6cb7968a69.PNG)
--------------------------------------------------------------------------------
The function like this :
![8](https://user-images.githubusercontent.com/48154419/90977526-a11ba080-e578-11ea-9bc4-cc398840ef85.PNG)
-------------------------------------------------------------------------------

**Comment by iliasam** [comment]
Do you understand right, that you **must not** change APD voltage/Laser voltage/Compactor threshold during calibration data recording?   
You need to change received signal by slowly closing APD lens with something non-transparent. 
Distance should be > 1.5m.

**Comment by Anesthesia-RW** [comment]
Thank u , I got it wrong before, i will try again.

**Comment by Anesthesia-RW** [comment]
Hello.
Following the steps you gave, I selected the constant distance is 3m and change received signal by slowly closing APD lens with something non-transparentcollect . 
The data and curves are as follows:
![11th_hi](https://user-images.githubusercontent.com/48154419/91327455-1f39aa80-e7f8-11ea-9890-ad4a8c8aca39.PNG)
![11th_cr](https://user-images.githubusercontent.com/48154419/91327460-206ad780-e7f8-11ea-9e69-12c67c85577d.PNG)
I sent the  "coff." to MCU, then change the distance to test.
Some data are as follows:
Real_Distance_1--- 5.15m     Test_Distance_1--- 5.95m
Real_Distance_2--- 4.55m     Test_Distance_2--- 5.04m
Real_Distance_3---2.35m      Test_Distance_3---1.95m
Real_Distance_4---1.75m      Test_Distance_4---1.01m
Real_Distance_5---1.2m        Test_Distance_5---0.16m
I found that when Real_Distance<1.1m the Test_Distance is 0.
There a problem with the calibration data. I repeated the calibration more than ten times. Sending  the calibration coefficient  to MCU and test ,the measurement result was quite different.
I use oscilloscope tested J1 and J2,The result is as follows:
--Distance:3m
![J1_TIA](https://user-images.githubusercontent.com/48154419/91330777-75a8e800-e7fc-11ea-8f00-612d525e19b8.png)
![J2_laser](https://user-images.githubusercontent.com/48154419/91330778-75a8e800-e7fc-11ea-81ab-061d75eb5ffd.png)

**Comment by iliasam** [comment]
Your results are looking strange.
Firstly - I can see that your APD voltage is 63V while target voltage is 128V. This is not normal.  What is the type of your APD?
Are you using the same optics as I am using in my Lidar?

Your calibration data are looking OK, so you can move "Fit Start" line to the 300-350 position before starting "Fit".

Also, after calibration you need to set: " "Ref distance" - distance to a reference object. Used with non-scanning firmware only. You need to enter current distance to the object and press "SET" button". Best distance is 2-3 m.
See https://github.com/iliasam/OpenTOFLidar/wiki/LidarTestingUtility

**Comment by Anesthesia-RW** [comment]
Because I can’t find the same APD as yours, no supply. So I changed the type of APD. The optics is the same as yours.

 I have two type APD.  One is MTAPD-07-14 ,the other one is GD5210Y-2 , all of them the APD voltage in range 63~65 (v).
Their performance is very similar to yours MTAPD-07-13

Especially MTAPD-07-14,I checked the information and found that it came from the same data sheet with  MTAPD-07-13.
The GD5210Y-2's data sheet as follows:
[996076217905nm系列APD-规格书.pdf](https://github.com/iliasam/OpenTOFLidar/files/5134165/996076217905nm.APD-.pdf)

**Comment by iliasam** [comment]
According to MTAPD-07-14 datasheet its breakdown voltage is in "160-200V" range.
You can see that APD is in breakdown state by seeing a very big noise at the TIA output.
  
According to GD5210Y-2 datasheet its breakdown voltage is in 130-220 V range.
  
In any case, it is **not OK** that measured APD voltage (63V) is not the same as set target voltage (128V). It is look like APD step-up converter or APD voltage measurement circuit are working wrong. 

You should **ensure** that the is no light is getting directly from laser to the APD (or APD lens) - by some holes in the lens holders or by scattering from mirror or mechanics.
I advice you to do calibration and all not scanning tests **without mirror** - lenses should be directly pointed to the target object. I mean this setup: https://habrastorage.org/getpro/habr/post_images/120/c20/e43/120c20e43dc0d41ada95b6ed9ecd82bf.jpg

**Comment by Anesthesia-RW** [comment]
I sorry, I forgot to tell you that the earlier calibration are under without mirror conditions .
I mean all test as you advise, constant distance is 3m and without mirror.

![IMG_20200827_150810](https://user-images.githubusercontent.com/48154419/91409900-b21a2980-e878-11ea-9a32-1c5756401f04.png)

**Comment by Anesthesia-RW** [comment]
Could it be a problem with my electronic components?
But all I follow your BOM table, maybe I need to check again and make a new PCB board.
I have made three PCB boards, one of the TIA is damaged. the voltage of the other two boards is about 63v.

**Comment by iliasam** [comment]
You cant try to switch to the manual PWM controlling by setting "APD PWM Steps" - second "SET" buttom at the "Rangefinder State" group. Try to change this value starting from low (like 30 steps) to bigger value.

**Comment by Anesthesia-RW** [comment]
Change the ”APD PWM“ and recording "APD Volt", the curve as follow:
When the APD PWM >85 , Changing the APD PWM  and the APD volt  changes very little. Fluctuate around 64
![myplot_2](https://user-images.githubusercontent.com/48154419/91425805-388d3600-e88e-11ea-9859-e5728fe988e0.png)

**Comment by iliasam** [comment]
This is very strange. Have you tried to measure real APD voltage by voltmeter?
Could you remove R1 (disconnect APD from power) and test how voltage is changing from "APD PWM" ?

**Comment by Anesthesia-RW** [comment]
I can't remove R1. 
Using voltmeter to measure the real APD voltage. I found the voltmeter's vlaue less than soft_APD voltage(MCU get)

**Comment by iliasam** [comment]
"I can't remove R1."
Why?
" I found the voltmeter's vlaue less than soft_APD voltage"
How big is the difference?

**Comment by Anesthesia-RW** [comment]
![myplot_1](https://user-images.githubusercontent.com/48154419/91538432-6afc6900-e94a-11ea-960e-659c7689c3ca.png)

**Comment by Anesthesia-RW** [comment]
The difference is in range (1,2).

**Comment by Anesthesia-RW** [comment]
Just now.I removed R1 and test again no change in voltage.

**Comment by iliasam** [comment]
"The difference is in range (1,2)."
Do you mean volts? If it is right, is is bad - voltmeter have own resistance, an power  of the APD step-up converter is very low, so voltmeter create some voltage drop. Also, APD_voltage is not measured very accurately. 

I can't suggest why APD voltage did not grow bigger that 70V. It look like some HW problem in APD voltage source.

**Comment by Anesthesia-RW** [comment]
Yes,the difference is  1v to 2 v.
Thanks for your reply.
I decided to remake a few PCB boards.
How to explain that the previous two pieces have the same error? 
Maybe problems with electronic components?

**Comment by iliasam** [comment]
"Maybe problems with electronic components?"
Yes, looks like that some of the components is damaged or wrong.
Are you sure that max voltage parameter of the capacitors C10/C11/C33  is 250V ? 
Is L5 inductance really 47uH ?

I don't think that there are problems with PCB itself.

**Comment by Anesthesia-RW** [comment]
I am also troubled by this problem. The electronic components is provided by a specialized manufacturer.

**Comment by iliasam** [comment]
But are you sure that **your** BOM (that you send to the components seller) is correct?

**Comment by Anesthesia-RW** [comment]
Hi，The APD power is generate by this  part  circuit?
![myplot_2](https://user-images.githubusercontent.com/48154419/91630746-904ead00-ea06-11ea-80a5-8bee0769c0a1.PNG)

**Comment by iliasam** [comment]
Yes.
R8 and R9 are used for measurement voltage by MCU.

**Comment by Anesthesia-RW** [comment]
I want to simulate this part of the circuit separately. 
The PWM signal is generated by a signal generator, but I cannot find L4 in Multisim. Can you give some suggestions?

**Comment by iliasam** [comment]
L4 is not needed for simulation. It is used to block noise from this step -up circuit.
APD_PWM is 100kHz.

**Comment by Anesthesia-RW** [comment]
I tested the APD with a voltmeter, but it showed only 4.71v,
![Multism](https://user-images.githubusercontent.com/48154419/91631340-6a77d700-ea0b-11ea-8fe4-6d1b4189e21f.PNG)

**Comment by Anesthesia-RW** [comment]
The PWM config(test by Oscilloscope):
frequency:100khz
duty:11%---->about 80steps
amplitude: 3.76v
![IMG_20200829_152831](https://user-images.githubusercontent.com/48154419/91631496-74e6a080-ea0c-11ea-9f01-fffe380548f5.png)

**Comment by iliasam** [comment]
![sim1](https://user-images.githubusercontent.com/10147086/91633856-41981780-e9f4-11ea-8d97-76628de3da3c.png)
I tried to run simulation too.
I found that Multisim BSS131 model is not working, so I changed it to the BSS123. Its is limited by 100V, so simulation does not give voltage bigger than 100V.
Real BSS131 is limited by 240V.
Virtual oscilloscope  slow down simulation, I advice you to remove it.

**Comment by Anesthesia-RW** [comment]
Great! 
Have you tested the relationship curve between APD_PWM and APD_PWM, as I posted before.

Hope my new PCB boards won’t go wrong!

**Comment by iliasam** [comment]
Do you mean - tested in real hardware?
I never tried to plot such curves, but I never have a problem to get APD voltage > 100V.
My APD power source circuit is always working OK.
You can see the same schematic here - https://cdn.hackaday.io/files/1635016995226784/Unruly_Rev1.pdf (see left bottom corner)

"Hope my new PCB boards won’t go wrong!"
Have you made any changes in PCB? Have you changed BOM?
Are you soldering components by yourself?

**Comment by Anesthesia-RW** [comment]
I mean--APD_Voltage_soft.

No changes.  I just ctrl_c and ctrl_v the file, then send to the seller.
I did not solder the components myself. 

Just now,i found a strange phenomenon. the APD_voltage equal about 102*R9_voltage, fit well with the source code.
This means that the voltage amplification function is normal, 
but what limits the maximum voltage of my APD to only 70v?

Changing the PWM steps APD voltage will also change, so the PWM input is normal, 
That’s why I asked if you have tested the relationship curve between APD and PWM. I want to know if it’s just that the maximum value of my APD voltage is limited, not the relationship.

**Comment by iliasam** [comment]
![APD_Voltage_vs_PWM_steps_PCB_v1](https://user-images.githubusercontent.com/10147086/91643488-50a2b800-ea3c-11ea-8a8b-04d6c52fc8ae.png)
I have made measurements with PCB v1 (AD500-8). Laser was closed. Small amount of external light. PWM controlling is manual.

You can see that when APD voltage > 90V voltage stop rising. This is breakdown mode - APD current begin to grow, and step-up efficiency fall. Noise level became very big at this voltage - "Max diff" value (at the center text window) will be very big (bigger that 1000).
"Max diff" value at 88V (not breakdown mode) is near 0-2 BINs (with closed laser).

As I know from PCB v4, that MTAPD-07-13 have higher breakdown voltage - ~120V, so it need more PWM steps.

So, if you want to test how good is APD voltage source, you need to remove R1 to disconnect APD from power.

************************************
"Just now,i found a strange phenomenon. the APD_voltage equal about 102*R9_voltage, fit well with the source code."
This just mean that voltage **measurement** part is working OK. This part consist of two resistors so there are no reasons for it to work wrong.
Problem is in step-up part.
APD breakdown voltage limiting (as I described earlier) is not seems your problem - you said that you have such problem when R1 was removed - no APD, no breakdown.
Also I can't see that terrible breakdown noise at your screenshots. 

Maybe Q1 (BSS131) is wrong? Try to by it somewhere else and replace.

*********************************
"No changes. I just ctrl_c and ctrl_v the file, then send to the seller."
Why do you think that something will change in this case?

**Comment by Anesthesia-RW** [comment]
ok, I will follow your advice.

If the Q1 is wrong. the APD_voltage will not exceed 5v,in my simulation it will stable at 4.701v. 
But now my APD_voltage is about 64v, could i say the Q1 is working OK?

yeah! I know there have somthing change in my case.That's why my APD_voltage is wrong.

**Comment by iliasam** [comment]
BSS131 model in Modelsim is strange, it does not work **at all**. You can see that simulated transistor have two additional pins at the right:
![image](https://user-images.githubusercontent.com/10147086/91653304-0c4d0180-eaa8-11ea-8279-452b8a7bcc75.png)
maybe they should be connected to something.
So try to change **simulated** transistor to BSS123.

**Comment by Anesthesia-RW** [comment]
You have already mentioned this problem, and i also changed the  simulated transistor to BSS123.
The simulation results show that the APD voltage will change with the change of the PWM duty cycle and can reach a high voltage.
Just like your  simulation:
https://user-images.githubusercontent.com/10147086/91633856-41981780-e9f4-11ea-8d97-76628de3da3c.png

Assuming that the Q1 is broken ---We can treat Q1(a high-speed switch) as a simple switch,only have two state OFF or ON
No  matter the wrong of Q1 is OFF or ON, the APD voltage will not exceed 5V.
However, my APD voltage can reach 64V, and it can change with the change of PWM.That means the Q1 is working Ok.

Can you change the APD voltage to 70V,?
And then observe whether it can work and  the distance measurement is no problem.

**Comment by iliasam** [comment]
"Assuming that the Q1 is broken"
I don't think that it is broken. It could be just not needed BSS131, but another transistor model - by mistake of someone who is buying BOM or assembling PCB/ or remarked transistor.

"Can you change the APD voltage to 70V,?
And then observe whether it can work and the distance measurement is no problem."
I'm sorry, I don't have time for that.

**Comment by Anesthesia-RW** [comment]
Thanks.

**Comment by iliasam** [comment]
I found some time and made a measurements with PCB v4 and APD voltage 65V (nominal is 125V).
I made a calibration - it is needed after changing APD voltage.
![image](https://user-images.githubusercontent.com/10147086/91757536-23d6d800-ebd7-11ea-9f54-abb1c466de5e.png)
I found that only a certain "Fit start" slider position is working OK (absolute distance measurement error was <5 cm).
I made accurate measurements only in 20-180 cm range; 3m looked good, but did not confirm it.

When I tried to move  "Fit start" left, absolute measurement quality fall (absolute distance measurement error was >10 cm):
![image](https://user-images.githubusercontent.com/10147086/91757876-b8d9d100-ebd7-11ea-9280-cd0d62594396.png)

I can't guaranty that lidar will work OK it such mode - you can see that fit (calibration) is working not well. 

P.S. You can try to install something nontransparent between lenses before calibration - to minimize laser light scattering or install long (>8 cm)  nontransparent (non metallic)  tube to the laser lens.

**Comment by Anesthesia-RW** [comment]
Great!
Very valuable information.I will do it. 

Through simulation and testing, I currently determine that the abnormal APD voltage may be due to the value of L5. If the value of L5 is not enough, the inductor energy storage will be insufficient. When the PWM duty cycle is increased, the APD voltage will not go up.

**Comment by Anesthesia-RW** [comment]
Hello.The APD voltage's problem solved. The problem is the inductor L5 type not value. 
The manufacturer gave a multilayer inductor, but the actual winding type. There is no description about L5 type in the BOM.

Then I tested again.The result is still not good.
Carlibration's  distance: 1m
Object: White wall
![page1](https://user-images.githubusercontent.com/48154419/92838938-f75e5f80-f411-11ea-96e4-dbe952ec430b.png)
The carlibration as follow, and the curve looks good.
![calibration_2](https://user-images.githubusercontent.com/48154419/92841061-6c329900-f414-11ea-83d2-c04ed5acdb20.png)

After calibration，i change the distance to testing. 
When the distance is within the range of 0.2m to 1m. The result is good.
The distance < 0.2, result is bad---64m.
When the distance is greater than 1m, the error will increase as the distance increases.
Ture distance:3m-----test result:4m-----error:1m.
Ture distance:1.7m---test result:2m-----error:30cm.

I used an oscilloscope to observe the output in some places（Laser，Laser_comp,TIA,APD_comp）
I think the Laser's output is strange.Big difference from you
1. 
**Laser：**
![laser](https://user-images.githubusercontent.com/48154419/92841097-75236a80-f414-11ea-9bc1-97c58c99c5ac.png)
2.
**Laser comp:**
![laser_comp](https://user-images.githubusercontent.com/48154419/92841153-8c625800-f414-11ea-8cf6-8097b31e406a.png)
3.
**TIA:**
![TIA](https://user-images.githubusercontent.com/48154419/92841189-971ced00-f414-11ea-8d2b-cf38b1556198.jpg)
4.
**APD comp:**
![APDcomp](https://user-images.githubusercontent.com/48154419/92841216-a00dbe80-f414-11ea-9709-109135e49071.png)

**Comment by iliasam** [comment]
"I think the Laser's output is strange.Big difference from you"
Are you using "oscilloscope ground spring" like here: [LINK](https://electronics.stackexchange.com/questions/136123/how-do-you-attach-an-oscilloscope-ground-spring)  ?

Do you set reference  distance ("Ref distance" value and its "SET" button)?

I can see from your calibration data that "Begin X" is 537. You can see that mine was near 400.
Try to close lens stronger during calibration data capture - you need more data with partially closed lens.

"When the distance is greater than 1m, the error will increase as the distance increases.
Ture distance:3m-----test result:4m-----error:1m.
Ture distance:1.7m---test result:2m-----error:30cm."
It is strange.
Could you give me screenshots of the utility for distances 0.5, 1m, 1.5m, 2m, 2.5m, 3m, bigger distances would be good too. 
You don't need to set distance to the object exactly (round numbers), but I need exactly measured real distances.

**Comment by Anesthesia-RW** [comment]
The oscilloscope ground like this:
![Os_grand](https://user-images.githubusercontent.com/48154419/92911318-5dbba000-f45b-11ea-8db4-bad7fa12b990.jpg)
And i use the oscilloscope to Observed the APD_PWM. Like this:
![PWM](https://user-images.githubusercontent.com/48154419/92911994-f81be380-f45b-11ea-8365-6f390d8797fd.jpg)

Real distances:0.5m
screenshots: 
![test_0_5](https://user-images.githubusercontent.com/48154419/92912923-db33e000-f45c-11ea-9de2-1a01fe51d42b.PNG)

Real distances:1m
screenshots: 
![page_1](https://user-images.githubusercontent.com/48154419/92913005-f0a90a00-f45c-11ea-93d6-91bb54d9cef1.PNG)
Real distances:1.5m
screenshots: 
![ca_1](https://user-images.githubusercontent.com/48154419/92913050-fd2d6280-f45c-11ea-8e21-19992bdcacc2.PNG)
Real distances:2m
screenshots: 
![test_2](https://user-images.githubusercontent.com/48154419/92913072-04ed0700-f45d-11ea-8e37-8e3981529e0a.PNG)
Real distances:2.5m
screenshots: 
![test_2_5](https://user-images.githubusercontent.com/48154419/92913094-0c141500-f45d-11ea-9e3d-7bd6d897f8bc.PNG)
Real distances:3m
screenshots: 
![test_3](https://user-images.githubusercontent.com/48154419/92913136-146c5000-f45d-11ea-8c66-5a4bdff2f8f8.PNG)
Real distances:3.5m
screenshots: 
![test_3_5](https://user-images.githubusercontent.com/48154419/92913171-1c2bf480-f45d-11ea-9272-41f27a835e3d.PNG)
Real distances:4m
screenshots: 
![4](https://user-images.githubusercontent.com/48154419/92913203-23eb9900-f45d-11ea-8013-f551f6c62e74.PNG)

Could  you use a short video to record the process of your calibration  operation?
I found when i record the calibration data. Among dozens of sets of data, only one set of data is looks good(curve.

**Comment by Anesthesia-RW** [comment]
The oscilloscope like this:
![Os_grand](https://user-images.githubusercontent.com/48154419/92915471-2f3fc400-f45f-11ea-98b2-6913d13095a5.jpg)
 
Yes, after i send the coeffA/B to MCU. i set the reference distance  and save to the flash.
Real distance: 0.5m
screenshots:
![test_0_5](https://user-images.githubusercontent.com/48154419/92915973-a8d7b200-f45f-11ea-82d3-cbeee118b463.PNG)

Real distance: 1m
screenshots:
![test_1m](https://user-images.githubusercontent.com/48154419/92916078-c442bd00-f45f-11ea-9c53-eb3eb726a66a.PNG)

Real distance: 1.5m
screenshots:
![page_1](https://user-images.githubusercontent.com/48154419/92916196-dae91400-f45f-11ea-9940-2deb2a5baafb.PNG)
Real distance: 2m
screenshots:
![test_2](https://user-images.githubusercontent.com/48154419/92916253-e9cfc680-f45f-11ea-822d-a52a3a9380c6.PNG)
Real distance: 2.5m
 screenshots:
![test_2_5](https://user-images.githubusercontent.com/48154419/92916299-f48a5b80-f45f-11ea-8748-7f761935c004.PNG)
Real distance: 3m
screenshots: 
![test_3](https://user-images.githubusercontent.com/48154419/92916346-00761d80-f460-11ea-89ae-20da587a7dc6.PNG)
Real distance: 3.5m
screenshots:
![test_3_5](https://user-images.githubusercontent.com/48154419/92916384-079d2b80-f460-11ea-8c9e-a2a948746a1b.PNG)
Real distance: 4m
screenshots:
![4](https://user-images.githubusercontent.com/48154419/92916426-11bf2a00-f460-11ea-9a6f-aa0000f0d45d.PNG)

Could you use a short video to record the process of your calibration operation?
I found when i record the calibration data. Among dozens of sets of data, only one set of data is looks good(curve

**Comment by iliasam** [comment]
You should use ground spring probe of UFL-BNC cable to get proper results.
Ground probe that you showed, is not suitable for high frequencies.

I would try to analyse results that you send in the near future.

"Could you use a short video to record the process of your calibration operation?"
I'm not sure that I have time for that.
I can just say that you need to get at least 40 captured points.

**Comment by iliasam** [comment]
Do you have an ability to rebuild code by yourself?
It's look like your TDCs BIN length is significantly change from mine value.

**Comment by Anesthesia-RW** [comment]
Do you mean this part of the code?
![src_code](https://user-images.githubusercontent.com/48154419/92984685-80db6380-f4de-11ea-9eb3-5db902563073.png)

**Comment by iliasam** [comment]
Yes, line 17.

**Comment by Anesthesia-RW** [comment]
I did not change it.

 According to the TDC-GP2 datasheet.The Bin is 45ps ---Length: 13.5mm.
Did i change it to 13.5mm?

**Comment by iliasam** [comment]
I have analysed data from your screenshots, it looks like  this value for your rangefinder should be 11.3f

**Comment by Anesthesia-RW** [comment]
OK. Thanks.
Could you told me how to analysed the data from screenshots?
The calibration formula is like this:

![公式](https://user-images.githubusercontent.com/48154419/92992172-94f18600-f51b-11ea-948a-434af7bd7096.PNG)

**Comment by iliasam** [comment]
I'm not sure that my analyse is correct. You need to try change the coefficient and tell me results.

**Comment by Anesthesia-RW** [comment]
Yeah！ your analyse is correct. The result is better than before.
Could you told me how to analysed the data from screenshots  to get the bin_length?  
The  calibration's  distance:1m
 calibration's screenshots like this:
![ca_1](https://user-images.githubusercontent.com/48154419/93019183-0f97cf80-f608-11ea-849f-39fcef8bec9c.PNG)
![pa_1](https://user-images.githubusercontent.com/48154419/93019189-1888a100-f608-11ea-8371-3e524d365e79.PNG)

Real distance: 0.5m
screenshots:
![pa_0_5](https://user-images.githubusercontent.com/48154419/93019250-61405a00-f608-11ea-8033-e061b63cbe65.PNG)


Real distance: 2m
screenshots:
![pa_2](https://user-images.githubusercontent.com/48154419/93019232-4e2d8a00-f608-11ea-9177-796e78a7a276.PNG)

Real distance: 2.5m
screenshots:
![pa_2_5](https://user-images.githubusercontent.com/48154419/93019257-6bfaef00-f608-11ea-80e1-d05174dd12b4.PNG)

Real distance: 3m
screenshots:

Real distance: 3.5m
screenshots:
![pa_3_05](https://user-images.githubusercontent.com/48154419/93019304-c1370080-f608-11ea-85d1-a67c514fecbc.PNG)
Real distance: 4m
screenshots:
![pa_4](https://user-images.githubusercontent.com/48154419/93019305-c85e0e80-f608-11ea-9c28-faae824c9fe0.PNG)
Real distance: 4.5m
screenshots:
![pa_4_5](https://user-images.githubusercontent.com/48154419/93019315-cf851c80-f608-11ea-8311-af246734f359.PNG)
Real distance: 5m
screenshots:
![pa_5](https://user-images.githubusercontent.com/48154419/93019324-d90e8480-f608-11ea-8c7b-9b0f649da796.PNG)
Real distance: 5.5m
screenshots:
![pa_5 5](https://user-images.githubusercontent.com/48154419/93019327-e0359280-f608-11ea-9bcd-758cefa06e8a.PNG)
Real distance: 6m
screenshots:
![pa_6](https://user-images.githubusercontent.com/48154419/93019359-05c29c00-f609-11ea-8bd2-4744e1974374.PNG)

Real distance: 6.5m
screenshots:

![pa_6_05PNG](https://user-images.githubusercontent.com/48154419/93019425-897c8880-f609-11ea-8d6a-33052cf8f1b3.PNG)

**Comment by iliasam** [comment]
In fact, my analyse was simple enough - I just build a table from your "Corr. AVR TOF" values and real distances (in mm), plot a curve and do a linear fit:
![image](https://user-images.githubusercontent.com/10147086/93020757-8ffa0880-f5e7-11ea-9f00-3142ab0d623d.png)
You need to do a calibration before this analyse.

I can see that you still getting some error during measurements.
Have you tried to get more calibration points in width range 550-630 BIN? 
The smaller received signal (more closed lens) - the smaller is width.

**Comment by Anesthesia-RW** [comment]
I  wil do it . 
And i try to testing the scanning model. 
The screenshots  like this:
![00](https://user-images.githubusercontent.com/48154419/93021032-18da6980-f613-11ea-97a1-7cdcc7f6fcb3.PNG)
I checked the src :
![01](https://user-images.githubusercontent.com/48154419/93021074-53440680-f613-11ea-85f7-279b02c49282.PNG)
According to the state mask table----0x30 means:Mirror is stopped or encoder is not working and Mirror speed is too low or too high.
But i checked the encoder it can output rectangular wave.

**Comment by iliasam** [comment]
Are you sure that your encoder disc have exactly 65 holes?
Do you have closed zero  (index) hole at the encoder disc ?
https://ibb.co/FV6nBdF
https://ibb.co/2Fpfjx4

**Comment by Anesthesia-RW** [comment]
My disc is 50 holes and i  changed the code.
I am sorry, I just checked this zero mask and it is missing. I will fix it and try again.
Thanks for your reply.

**Comment by iliasam** [comment]
Do you have any results?

**Comment by Anesthesia-RW** [comment]
Yeah!  The lidar can capture many points. But  the  point cloud is not very good
I’m learning its principle recently, I want to improve it.
Your work is very fascinating.

**Comment by iliasam** [comment]
"But the point cloud is not very good"
What is wrong with the results?

**Comment by pikasm** [comment]
Hello, can you tell me where you bought your encoder and encoder? I'm trying to make this radar, and my radar has no result when the motor rotates.

**Comment by iliasam** [comment]
You are asking in a wrong closed issue...
https://github.com/iliasam/OpenTOFLidar/wiki/Encoder
"I have taken this disc from old ball mouse."
"I have taken mine sensor from old printer so I can't give exact name of it. It is conventional opto-interrupter, with one LED and one NPN photo-transistor."

**Comment by pikasm** [comment]
I'm sorry, I just saw him make it, and he successfully measured some cloud images when he was spinning, so I want to ask him. I have noticed that your encoder comes from the printer.
